### PR TITLE
Remove support for CoreOS and Jessie

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

**Comment by hakman** [comment]
This should also be removed:
https://github.com/kubernetes/kops/blob/d5e3e15377f24d29da6bd4c6a1e9ea2a9477561f/upup/pkg/fi/nodeup/nodetasks/package.go#L340-L344

File: nodeup/pkg/distros/identify.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func FindDistribution(rootfs string) (Distribution, error) {
 	if err == nil {
 		debianVersion := strings.TrimSpace(string(debianVersionBytes))
 		if strings.HasPrefix(debianVersion, "8.") {
-			return DistributionJessie, nil
+			return "", fmt.Errorf("distribution Jessie is no longer supported")
```
[the remark this answers is not in the document]
Comment (johngmyers) [reply nodeup/pkg/distros/identify.go:54]: I think you mean 8.

[the remark this answers is not in the document]
Comment (hakman) [reply nodeup/pkg/distros/identify.go:54]: Yes 😄

**Comment by hakman** [comment]
I think this can also be removed:
https://github.com/kubernetes/kops/blob/759e24a4905dd190bf7de648cafd991a40156944/pkg/apis/kops/validation/aws.go#L103-L107

**Comment by rifelpet** [comment]
Nice work. I wonder if theres anything we can do to surface the error earlier, so that users that don't read the release notes aren't confused why their `rolling-update cluster` is failing to validate  after the first node is being replaced.

I see `awsValidateAMIforNVMe` was using the name of the image which is far from complete coverage but I wonder if doing something similar would be "good enough" coverage to prevent the majority of support inquiries.

**Comment by hakman** [comment]
@rifelpet we have similar thing to instance type validation there. IMHO, we should do it in apply, where we can get the image name and owner account even for the AMI. Still I would do it in a subsequent patch.

**Comment by johngmyers** [comment]
@justinsb this PR is intended for 1.18. We shouldn't support these EOL distros for 1.18.

**Comment by hakman** [comment]
I think that we agreed during last office hours to include this in 1.18. Should this be merged as is, or there are any other issues to be addressed?

**Comment by rdrgmnzs** [comment]
Do we want to remove the references in the two files bellow in a separate PR?
kops/upup/pkg/fi/nodeup/nodetasks/service.go
cmd/kops/update_cluster.go

There's also a few references to CoreOS in the docs that we may want to cleanup separately to minimize any confusion.

Otherwise this LGTM.

**Comment by johngmyers** [comment]
Digital Ocean still defaults to a CoreOS image.

openstack.md says it defaults to CoreOS, but I don't see any code that makes this so.

Someone else will have to tackle docs/examples/coreos-kops-tests-multimaster.md

**Comment by hakman** [comment]
https://github.com/kubernetes/kops/pull/9181 will address the default image for DO.

**Comment by johngmyers** [comment]
The SIG-Storage flakes are strong today.
/retest

**Comment by hakman** [comment]
Feels like Jan all over again 😄

**Pull request merged**